foladipo/docs-cabinet-cp2
# Setup frontend dev environment

#### What does this PR do?
It sets up the development environment for the frontend of this app. This involved installing and setting up npm packages like react, react-router, redux and many others. Note, however, that some frontend functionality DID end up being implemented in order to test the dev environment. So, this PR has some of the work in #147026181. 

#### Description of task to be completed
Set up frontend dev environment.

#### How should this be manually tested?
*Required*: You need to have a Postgres database server running on your computer. Then, create a `.env` according to [this template](https://github.com/andela-foladipo/docs-cabinet-cp2/blob/feature/146115889/create-endpoints-using-tdd/.env.sample).

In a terminal/command prompt on your computer, switch to the `chore/147025811/setup-frontend-dev-environment` branch of this repo. Then:
- Run each of these commands:
```bash
npm install

npm run db:migrate

npm run db:seed
```

- You'll need to manually create an admin user though as we don't seed such a user for security reasons. To do this, we recommend [pgAdmin](https://www.pgadmin.org/). Don't forget to add the credentials of this account that are required in your `.env` file.

- Then, run `npm start`.

- Note the `PORT` you set in your `.env` file for use below. If you didn't set it, its default value is `5000`.

- Open `localhost:PORT` on your machine. You should see a (work-in-progress) log in page.

#### Any background context you want to provide?
N/A

#### What are the relevant Pivotal Tracker stories?
147025811, 147026181

#### Screenshots
N/A
